addons/portal/models/mail_thread.py

```diff
@@ -87,3 +92,15 @@ def _portal_get_parent_hash_token(self, pid):
         :return: False or logical parent's _sign_token() result
         """
         return False
+
+    @api.model
+    def _get_thread_with_access(self, thread_id, mode="read", **kwargs):
+        try:
+            return super()._get_thread_with_access(thread_id, mode, **kwargs)
+        except AccessError:
```

Comment:
This should not check for super raising, but returning empty recordset instead to have consistent API.